test/fetch/client-fetch.js

```diff
@@ -165,24 +165,83 @@ test('unsupported formData 1', (t) => {
   })
 })
 
-test('unsupported formData 2', (t) => {
+test('multipart formdata not base64', async (t) => {
+  // Construct example form data, with text and blob fields
+  const formData = new FormData()
+  formData.append('field1', 'value1')
+  const blob = new Blob(['example\ntext file'], { type: 'text/plain' })
+  formData.append('field2', blob, 'file.txt')
+
+  const tempRes = new Response(formData)
+  const boundary = tempRes.headers.get('content-type').split('boundary=')[1]
+  const formRaw = await tempRes.text()
+
+  const server = createServer((req, res) => {
+    res.setHeader('content-type', 'multipart/form-data; boundary=' + boundary)
+    res.write(formRaw)
+    res.end()
+  })
+  t.teardown(server.close.bind(server))
+
+  server.listen(0, () => {
+    fetch(`http://localhost:${server.address().port}`)
+      .then(res => res.formData())
+      .then(form => {
+        const field1 = form.get('field1')
+        t.equal(field1, 'value1')
+        const field2 = form.get('field2')
+        return field2.text()
+      })
+      .then(text => {
+        t.equal(text, 'example\ntext file')
+      })
+  })
+})
+
+test('multipart formdata base64', (t) => {
   t.plan(1)
 
+  // Example form data with base64 encoding
+  const formRaw = '------formdata-undici-0.5786922755719377\r\nContent-Disposition: form-data; name="key"; filename="test.txt"\r\nContent-Type: text/plain\r\nContent-Transfer-Encoding: base64\r\n\r\ndmFsdWU=\r\n------formdata-undici-0.5786922755719377--'
   const server = createServer((req, res) => {
-    res.setHeader('content-type', 'multipart/form-data')
+    res.setHeader('content-type', 'multipart/form-data; boundary=----formdata-undici-0.5786922755719377')
+    res.write(formRaw)
     res.end()
   })
   t.teardown(server.close.bind(server))
 
   server.listen(0, () => {
     fetch(`http://localhost:${server.address().port}`)
       .then(res => res.formData())
-      .catch(err => {
-        t.equal(err.name, 'NotSupportedError')
+      .then(form => form.get('key').text())
+      .then(text => {
+        t.equal(text, 'value')
       })
   })
 })
 
+test('busboy emit error', async (t) => {
+  const formData = new FormData()
+  formData.append('field1', 'value1')
+
+  const tempRes = new Response(formData)
+  const formRaw = await tempRes.text()
+
+  const server = createServer((req, res) => {
+    res.setHeader('content-type', 'multipart/form-data; boundary=wrongboundary')
+    res.write(formRaw)
+    res.end()
+  })
+  t.teardown(server.close.bind(server))
+
+  server.listen(0, async () => {
+    const res = await fetch(`http://localhost:${server.address().port}`)
+    res.formData().catch(err => {
+      t.equal(err.message, 'Unexpected end of form')
+    })
+  })
```

Comment:
Try something like this, treat like it like pseudocode:

```suggestion
  const { promisify } = require('util')
  const listen = util.promisify(server.listen.bind(server))
  await listen(0)
  const res = await fetch(`http://localhost:${server.address().port}`)
  await t.rejects(res.formData(), 'Unexpected end of form')
```